src/app/pages/connect-page/connection-details/connection-details.page.ts

```diff
@@ -38,7 +64,11 @@ export class ConnectionDetailsPage implements OnInit{
           text: "Oops, no..",
           role: 'cancel'
         }, {
-          text: 'Yes!'
+          text: 'Yes!',
+          handler: () => {
+            this._connectModelService.removeConnection(this.connectedWithUserId);
+            this._router.navigate(['/connect/list-connections']);
```

Comment:
There is a promise being returned from the model service call. Use it, so that you call the router until the model service call is done done.